From c56fe95d9dc59b4015e5ba1041cdc85e952ba42a Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Thu, 26 Oct 2023 11:44:48 +0200
Subject: [PATCH 380/954] feat(stm32mp2): invalidate cache after BL2 DT unmap

This memory will be used afterwards by BL31 data. To avoid cache issues,
invalidate this area. No need to clean, as it is first mapped as a
read-only memory.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git f6ad9c94383862dc6585b35e602cbdf86040f3c1 feat(stm32mp2): invalidate cache after BL2 DT unmap]
Change-Id: I8e0d48be314efee29e5d9e3f47ecfb6e30cae7f7
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/336433
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 plat/st/stm32mp2/bl2_plat_setup.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/plat/st/stm32mp2/bl2_plat_setup.c b/plat/st/stm32mp2/bl2_plat_setup.c
index eba386edae..4e50ee0421 100644
--- a/plat/st/stm32mp2/bl2_plat_setup.c
+++ b/plat/st/stm32mp2/bl2_plat_setup.c
@@ -473,6 +473,7 @@ int bl2_plat_handle_post_image_load(unsigned int image_id)
 		}
 #endif
 		mmap_remove_dynamic_region(DTB_BASE, DTB_LIMIT - DTB_BASE);
+		inv_dcache_range(DTB_BASE, DTB_LIMIT - DTB_BASE);
 
 		break;
 
-- 
2.53.0

